haskell/src/Monocle/Backend/Queries.hs

```diff
@@ -1066,11 +1004,34 @@ data MetricInfo = MetricInfo
 
 data Metric m a = Metric
   { metricInfo :: MetricInfo,
-    runMetric :: m a
+    runMetric :: m (Numeric a),
+    runMetricTrend :: m (V.Vector (Histo a))
   }
 
-instance Functor m => Functor (Metric m) where
-  fmap f x = Metric (metricInfo x) (f <$> runMetric x)
+instance (Functor m) => Functor (Metric m) where
+  fmap f x = Metric (metricInfo x) (convN <$> runMetric x) (convT <$> runMetricTrend x)
```

Comment:
If Numeric and Histo implement Functor, then we could remove the helper like this:

```suggestion
  fmap f x = Metric (metricInfo x) (fmap f <$> runMetric x) (fmap f <$> runMetricTrend x)
```

And using pattern match:

```suggestion
  fmap f Metric {..} = Metric metricInfo (fmap f <$> runMetric) (fmap f <$> runMetricTrend)
```